bento_etl/loaders/dependencies.py

```
@@ -2,21 +2,26 @@
 from typing import Annotated
 
 from bento_etl.loaders.base import BaseLoader
+from bento_etl.loaders.phenopackets_loader import PhenopacketsLoader
 from bento_etl.logger import LoggerDependency
 from bento_etl.models import Job
+from bento_etl.config import ConfigDependency
 
 __all__ = ["get_loader", "LoaderDep"]
 
 
-def get_loader(job: Job, logger: LoggerDependency):
+def get_loader(job: Job, logger: LoggerDependency, config: ConfigDependency):
     # returns the appropriate loader instance depending on the job description
     # TODO: implement dependency injection logic once we have Job model and concrete loaders
     # e.g:
     # if job.loader.type == "phenopacket":
     #   return LoadPhenopackets(job.loader)
 
     # TODO: should probably raise if dependency cannot be provided from job details
-    return BaseLoader(logger)
+    if job.loader.data_type == "phenopackets":
+        return PhenopacketsLoader(logger, config, job.loader.dataset_id)
+    
+    return BaseLoader(logger, config)
```

We should raise a NotImplemented error for jobs that present a data type we don't handle. The BaseLoader return was just as a placeholder.